internal/server/audit/kafka/kafka.go

```diff
@@ -0,0 +1,140 @@
+package kafka
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/sasl/plain"
+	"github.com/twmb/franz-go/pkg/sr"
+	"go.flipt.io/flipt/internal/config"
+	"go.flipt.io/flipt/internal/server/audit"
+	"go.uber.org/zap"
+)
+
+const (
+	sinkType      = "kafka"
+	encodingProto = "protobuf"
+	encodingAvro  = "avro"
+)
+
+// encodingFn represents func which encoding data.
+type encodingFn func(v any) ([]byte, error)
+
+// Encoder represents an interface for encoding data and provide the schema.
+type Encoder interface {
+	Schema() sr.Schema
+	Encode(v any) ([]byte, error)
+}
+
+type Sink struct {
+	logger *zap.Logger
+	client *kgo.Client
+	encode encodingFn
+}
+
+// NewSink is the constructor for a Sink.
+func NewSink(logger *zap.Logger, cfg config.KafkaSinkConfig) (audit.Sink, error) {
+	logger = logger.With(zap.String("sink", sinkType))
+	opts := []kgo.Opt{
+		kgo.SeedBrokers(cfg.BootstrapServers...),
+		kgo.DefaultProduceTopic(cfg.Topic),
+		kgo.WithLogger(&klogger{logger: logger}),
+	}
+
+	if cfg.Authentication != nil {
+		auth := plain.Auth{User: cfg.Authentication.Username, Pass: cfg.Authentication.Password}
+		tlsDialer := &tls.Dialer{}
+		opts = append(opts,
+			kgo.SASL(auth.AsMechanism()),
+			kgo.Dialer(tlsDialer.DialContext),
+		)
+	}
+
+	client, err := kgo.NewClient(opts...)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I'd like to suggest to make a _ping_ after creating the client:
```suggestion
	client, err := kgo.NewClient(opts...)
	if err != nil {
		return nil, err
	}
	
	if err := client.Ping(context.Background()); err != nil {
		return nil, err
	}
```

The reason is because the `NewClient` function only returns an error if an invalid config is provided, it does not test the connection with Kafka which can lead to an unclear unexpected behavior when dealing with auth

Reply:
@thepabloaguilar thank you for the great idea